core/src/ledger_cleanup_service.rs

```diff
@@ -0,0 +1,101 @@
+//! The `ledger_cleanup_service` drops older ledger data to limit disk space usage
+
+use crate::blocktree::Blocktree;
+use crate::result::{Error, Result};
+use crate::service::Service;
+use solana_sdk::pubkey::Pubkey;
+use solana_sdk::timing::DEFAULT_SLOTS_PER_EPOCH;
+use std::string::ToString;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::mpsc::{Receiver, RecvTimeoutError};
+use std::sync::Arc;
+use std::thread;
+use std::thread::{Builder, JoinHandle};
+use std::time::Duration;
+
+pub const DEFAULT_MAX_LEDGER_SLOTS: u64 = DEFAULT_SLOTS_PER_EPOCH * 5;
+
+pub struct LedgerCleanupService {
+    t_cleanup: JoinHandle<()>,
+}
+
+impl LedgerCleanupService {
+    #[allow(clippy::new_ret_no_self)]
```

Comment:
is the clippy warning necessary?

Reply:
Yeah this seems unnecessary I must have been returning other crap earlier.